docs/tutorial_history_backend.rst

```diff
@@ -0,0 +1,319 @@
+.. _tutorial_history_backend:
+
+****************************************
+Tutorial: Write Your Own History Backend
+****************************************
+
+One of best thing you can do with xonsh is that you could customize
```

Comment:
"One of the great things about xonsh is how easy it is to customize. In this tutorial, let's write our own history backed based on CouchDB."